text/3371-cargo-target-dir-templates.md

```diff
@@ -0,0 +1,411 @@
+- Feature Name: `cargo_target_dir_templates`
+- Start Date: 2023-01-12
+- RFC PR: [rust-lang/rfcs#3371](https://github.com/rust-lang/rfcs/pull/3371)
+<!-- - Cargo Issue: [rust-lang/cargo#0000](https://github.com/rust-lang/cargo/issues/0000) -->
+
+# Summary
+[summary]: #summary
+
+<!-- One paragraph explanation of the feature. -->
+
+Introduce templating to `CARGO_TARGET_DIR` to have `cargo` adapts its target directory dynamically depending on (at least) the manifest's path.
```

Comment:
typo:
 ```suggestion
Introduce templating to `CARGO_TARGET_DIR` to have `cargo` adapt its target directory dynamically depending on (at least) the manifest's path.
```

Consider expanding the summary somewhat:

> Introduce templating to [`CARGO_TARGET_DIR`](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget-dir).  The value `{manifest-path-hash}` is replaced with a hash based on the path to the current manifest.

Reply:
Done in https://github.com/rust-lang/rfcs/commit/05b96f8c6638b4bdb72ad240a0e8f8098715f7f6